fontforgeexe/fontview.c

```diff
@@ -850,23 +867,26 @@ return( sf->changed );
 }
 
 static int _FVMenuClose(FontView *fv) {
-    int i;
+    int filesaved, scriptsaved;
     SplineFont *sf = fv->b.cidmaster?fv->b.cidmaster:fv->b.sf;
 
     if ( !SFCloseAllInstrs(fv->b.sf) )
 return( false );
 
+    if ( warn_script_unsaved && fv->script_unsaved ) scriptsaved = AskScriptChanged();
+
     if ( fv->b.nextsame!=NULL || fv->b.sf->fv!=&fv->b ) {
 	/* There's another view, can close this one with no problems */
     } else if ( SFAnyChanged(sf) ) {
-	i = AskChanged(fv->b.sf);
-	if ( i==2 )	/* Cancel */
+	filesaved = AskChanged(fv->b.sf);
```

Comment:
Why bother changing all this logic vs. just returning on line 876 if AskScriptChanged returns no?

Reply:
This way if both the file and the script are unsaved, both get warned about. Perhaps they would both get warned about eventually, but if you have to return to FontForge, best for user to do everything forgotten at once. I also think this way makes the order of execution and asking much clearer, and the new variable names are better.

Reply:
Personally I'd find this behaviour surprising and counter-intuitive; if I've already been prompted to cancel closing the window and I cancel it, and then get asked *again* if I want to cancel, by my regard it's as if it's ignored what I chose earlier. I think it'd be much more 'normal' if it exits as soon as it knows it can.

There's also a problem with your changed logic. Supposing SFAnyChanged is false, `filesaved` will be uninitialised, which means you'll be comparing against an uninitialised value. Also I don't know if it matters, but you're now always calling `SFClearAutoSave` even if `SFAnyChanged` is false.

Reply:
OK, no problem, I don't really have a strong opinion on this, so just reverted most of the changes to this function in https://github.com/fontforge/fontforge/pull/3852/commits/0a5629ac1f9f165e30e873a09f3a7dc6f98a3916